### PR TITLE
Debug Trace Read: Fix trace filtering by service

### PR DESCRIPTION
**Pull request by crufter**
API and implementation changes too.

In support of https://github.com/micro/platform/pull/14

### CODE REVIEW

File: debug/debug.go [hunk 1]
```diff
@@ -85,10 +85,11 @@ func Run(ctx *cli.Context, srvOpts ...micro.Option) {
 	}()
 
 	// stats handler
-	statsHandler, err := statshandler.New(done, ctx.Int("window"))
+	statsHandler, err := statshandler.New(done, ctx.Int("window"), service)
```
Comment (crufter) [inline debug/debug.go]: This is actually pointless and just a leftover from testing, removing.

File: debug/trace/handler/handler.go [hunk 2]
```diff
@@ -46,6 +47,56 @@ type Trace struct {
 	cached              []*registry.Service
 }
 
+// Filters out all spans that are part of a trace that hits a given service.
+func filterSpansHittingAService(service string, snapshots []*trace.Snapshot) []*trace.Span {
```
Comment (asim) [inline debug/trace/handler/handler.go]: filterServiceSpans?

File: debug/trace/proto/trace.proto [hunk 3]
```diff
@@ -23,8 +23,9 @@ message Node {
 	string address = 2;
 }
 
-// Snapshot is a snapshot of Debug.Stats from a particular service when called.
+// Snapshot is a snapshot of Trace.Read from a particular service when called.
 message Snapshot {
+	// Source of the service where the snapshot was collected from
 	Service service = 1;
 	// Unix timestamp, e.g. 1575561487
     repeated Span spans = 2;
```
Comment (asim) [inline debug/trace/proto/trace.proto]: little formatting issue

File: debug/trace/handler/handler.go (hunk 2 above)
Comment (crufter) [reply debug/trace/handler/handler.go]: Works for me!

File: debug/trace/proto/trace.proto (hunk 3 above)
Comment (crufter) [reply debug/trace/proto/trace.proto]: Ahh!

**Comment by asim** [comment]
Self merge when you are ready!

**Pull request merged**